helper-cli/src/main/kotlin/commands/VerifySourceArtifactCurationsCommand.kt

```diff
@@ -0,0 +1,127 @@
+)
+internal class VerifySourceArtifactCurationsCommand : CommandWithHelp() {
+    @Parameter(
+        description = "A YAML file that contains package curation data.",
+        names = ["--package-curations-file"],
+        order = PARAMETER_ORDER_MANDATORY
+    )
+    private lateinit var packageCurationsFile: File
+
+    override fun runCommand(jc: JCommander): Int {
+        var success = true
+        val curations = packageCurationsFile.readValue<List<PackageCuration>>()
+
+        curations.forEach { curation ->
+            curation.data.sourceArtifact?.let { sourceArtifact ->
+                println("\n-----")
+                println("Checking source artifact for ${curation.id.toCoordinates()}.")
+                println("Downloading ${sourceArtifact.url}.")
+                val file = try {
+                    download(sourceArtifact.url)
+                } catch (e: Exception) {
+                    println("Failed to download source artifact: ${e.collectMessagesAsString()}")
+                    success = false
+                    null
+                }
+                if (file == null) {
+                    println("Aborting verification.")
+                } else {
+                    val hash = file.hash(sourceArtifact.hash.algorithm.aliases.first())
+                    println("Expected hash: ${sourceArtifact.hash.algorithm} ${sourceArtifact.hash.value}")
+                    println("Actual hash  : ${sourceArtifact.hash.algorithm} $hash")
+                    val verified = sourceArtifact.hash.verify(file)
+                    if (!verified) {
+                        println("Hashes do NOT match!")
+                        success = false
+                    }
+                }
+            }
+        }
+
+        println("\n-----")
+        return if (success) {
+            println("Successfully verified all source artifact curations.")
+            0
+        } else {
+            println("Some source artifact curations could NOT be verified, check the log for details.")
+            1
+        }
+    }
+
+    private fun download(url: String): File? {
+        val request = Request.Builder()
+            // Disable transparent gzip, otherwise we might end up writing a tar file to disk and expecting to
+            // find a tar.gz file, thus failing to unpack the archive.
+            // See https://github.com/square/okhttp/blob/parent-3.10.0/okhttp/src/main/java/okhttp3/internal/ \
+            // http/BridgeInterceptor.java#L79
+            .addHeader("Accept-Encoding", "identity")
+            .get()
+            .url(url)
+            .build()
+
+        try {
+            OkHttpClientHelper.execute(HTTP_CACHE_PATH, request).use { response ->
+                val body = response.body
+                if (!response.isSuccessful || body == null) {
+                    println("Failed to download source artifact: $response")
```

Comment:
`Failed to download source artifact: $response"` this functions should not speak of source artifacts because it just deals with URLs

Reply:
Done.